src/d3a/models/market/market_structures.py

```diff
@@ -115,6 +148,20 @@ def _to_csv(self):
         rate = round(self.offer.price / self.offer.energy, 4)
         return self[:2] + (rate, self.offer.energy) + self[3:5]
 
+    def to_JSON_string(self):
+        trade_dict = self._asdict()
+        trade_dict['offer'] = trade_dict['offer'].to_JSON_string()
+        trade_dict['residual'] = trade_dict['residual'].to_JSON_string() \
+            if trade_dict['residual'] is not None else None
+        trade_dict['time'] = trade_dict['time'].isoformat()
+        return json.dumps(trade_dict)
+
+
+def trade_from_JSON_string(trade_string):
+    trade_dict = json.loads(trade_string)
+    trade_dict['offer'] = offer_from_JSON_string(trade_dict['offer'])
+    return Trade(**trade_dict)
```

Comment:
@spyrostz Before you enter this rabbit hole, please add these lines to this function:
```
    if "residual" in trade_dict and trade_dict["residual"] is not None:
        trade_dict['residual'] = offer_from_JSON_string(trade_dict['residual'])
```
I will add this in my PR, too.